src/kaltura-player.js

```diff
@@ -119,28 +119,35 @@ class KalturaPlayer extends FakeEventTarget {
     this.reset();
     this._localPlayer.loadingMedia = true;
     this._uiWrapper.setLoadingSpinnerState(true);
-    const providerResult = this._provider.getMediaConfig(mediaInfo);
-    providerResult
-      .then(
-        (providerMediaConfig: ProviderMediaConfigObject) => {
-          const mediaConfig = Utils.Object.copyDeep(providerMediaConfig);
-          if (mediaOptions) {
-            mediaConfig.sources = mediaConfig.sources || {};
-            mediaConfig.sources = Utils.Object.mergeDeep(mediaConfig.sources, mediaOptions);
+    return new Promise((resolve, reject) => {
+      this._provider
+        .getMediaConfig(mediaInfo)
+        .then(
+          (providerMediaConfig: ProviderMediaConfigObject) => {
+            const mediaConfig = Utils.Object.copyDeep(providerMediaConfig);
+            if (mediaOptions) {
+              mediaConfig.sources = mediaConfig.sources || {};
+              mediaConfig.sources = Utils.Object.mergeDeep(mediaConfig.sources, mediaOptions);
+            }
+            const mergedPluginsConfigAndFromApp = mergeProviderPluginsConfig(mediaConfig.plugins, this.config.plugins);
+            mediaConfig.plugins = mergedPluginsConfigAndFromApp[0];
+            this._appPluginConfig = mergedPluginsConfigAndFromApp[1];
+            this.configure(getDefaultRedirectOptions(this.config, mediaConfig));
+            this.setMedia(mediaConfig);
+            return mediaConfig;
+          },
+          e => {
+            this._localPlayer.dispatchEvent(
+              new FakeEvent(CoreEventType.ERROR, new Error(Error.Severity.CRITICAL, Error.Category.PLAYER, Error.Code.LOAD_FAILED, e))
+            );
+            reject(e);
```

Comment:
what did we use to return as an error before?

Reply:
We returned the `providerPromise`, so if it was rejected, you returned rejected promise.

Reply:
and what was the object of it?

Reply:
I think we should keep the isolation we have today and return the error object with the error nested in it

Reply:
WDYT?

Reply:
> and what was the object of it?

undefined

> I think we should keep the isolation we have today and return the error object with the error nested in it

Good point, fixed.